File: updated_code/code/java/1356-sort-integers-by-the-number-of-1-bits.java
package com.leetcode;

import java.util.Arrays;

public class Solution {
    /**
     * Sorts the integers in the array in ascending order by the number of '1's in their binary representation
     * and in case of two or more integers have the same number of '1's, sorts them in ascending order.
     *
     * @param arr the input array
     * @return the array after sorting
     */
    public int[] sortByBits(int[] arr) {
        int[] result = new int[arr.length];
        System.arraycopy(arr, 0, result, 0, arr.length);
        Arrays.sort(result, (a, b) -> {
            int oneCountA = Integer.bitCount(a);
            int oneCountB = Integer.bitCount(b);
            if (oneCountA != oneCountB) {
                return Integer.compare(oneCountA, oneCountB);
            } else {
                return Integer.compare(a, b);
            }
        });
        return result;
    }
}

File: updated_code/code/python/1356-sort-integers-by-the-number-of-1-bits.py
Python
from typing import List

class SortIntegersByTheNumber:
    """
    This class is used to sort integers in an array in ascending order by the number of 1's in their binary representation.
    If two or more integers have the same number of 1's, they are sorted in ascending order.
    """

    def sort_by_bits(self, arr: List[int]) -> List[int]:
        """
        This method sorts the integers in the array in ascending order by the number of 1's in their binary representation.
        If two or more integers have the same number of 1's, they are sorted in ascending order.

        Parameters:
        arr (List[int]): The input array of integers.

        Returns:
        List[int]: The array after sorting it.
        """
        return sorted(arr, key=lambda x: (bin(x).count('1'), x))